app/controllers/sites_controller.rb

```diff
@@ -174,9 +174,11 @@ def edit
   # update sanitized params
   def update
     site_attributes = site_params
-    # though Site accepts nested attributes ofr nav_links for persistence, we want to handle the updates
+    # though Site accepts nested attributes of nav_links for persistence, we want to handle the updates
     # specially (to accommodate the deletion and reordering without recourse to record id)
     nav_links_attributes = site_attributes.delete('nav_links_attributes')
+    banner_upload = params[:site].delete(:banner)
```

Comment:
I'm a little hesitant to modify the `params` object's data since it can be accessed outside of this controller action.

Reply:
I understand (agree even!). The problem this is trying to address is keeping the file data out of the site_params, which (even though it is not permitted) it kept leaking into. This might be a Rails 4-ism?

Reply:
How would you feel about an alternate `site_params_without_file_data` method that internally calls `site_params` and runs the deletions, and returns the post-deletion object?

Reply:
I realize that's a bit inefficient though, since we'd be running the require/permit logic twice in this action by calling site_params and site_params_without_file_data.

Reply:
hmm, interesting - I tried having an independent params method, but not deleting from the filtered params in the block.

Reply:
pushed a minor change to do this

Reply:
I like it!  Thanks!